src/main/scala/rise/core/types/Nat.scala

```diff
@@ -5,41 +5,52 @@ import rise.core.types
 
 class NatIdentifier(
     override val name: String,
-    override val range: Range = RangeUnknown,
-    override val isExplicit: Boolean = false
+    override val range: Range,
+    override val isExplicit: Boolean,
+    val isTuningParam: Boolean
 ) extends NamedVar(name, range)
```

Comment:
@umazalakain any insight on how to translate this into the new identifier design?

Reply:
Would it be possible to keep a set of globally unbound identifier names that are meant to be tuned?

Reply:
Maybe, but there are some challenges:
1. Ideally we would keep a syntax similar to `tuningParam("vec", RangeAdd(0, 32, 1), (vec: Nat) => /* use vec ... */)` to create tuning parameters in an expression
2. Composing two expressions should also compose their set of tuning parameters (it doesn't really sound like a global set anymore)
3. Rewrite rules should be able to introduce tuning parameters using the same syntax